### PR TITLE
builtin/string.v: Speed up string.clone()

### PR DESCRIPTION
**Pull request by ntrel**
Use `memcpy` in `string.clone()`.
Also remove unnecessary null test (null is checked in `tos2`).
Make `cstring_to_vstring` just call `tos_clone()` - ~~this fixes copying a byte past the V string data (with `memdup`)~~ [that was just after my first commit!].

### CODE REVIEW

**Comment by medvednikov** [comment]
conflict after the `unsafe` PR merge

**Comment by ntrel** [comment]
Rebased. Can I make a pull to deprecate `cstring_to_vstring` in favour of `tos_clone`? The `_clone` part makes it clearer that a copy is returned.

File: vlib/builtin/string.v [hunk 1]
```diff
@@ -152,12 +147,7 @@ pub fn (s string) cstr() byteptr {
 
 // cstring_to_vstring creates a copy of cstr and turns it into a v string
 pub fn cstring_to_vstring(cstr byteptr) string {
-	unsafe {
-		slen := C.strlen(charptr(cstr))
-		mut s := byteptr(memdup(cstr, slen + 1))
-		s[slen] = `\0`
-		return tos(s, slen)
-	}
+	return tos_clone(cstr)
```
Comment (spytheman) [inline vlib/builtin/string.v:150]: This will create 2 v strings, instead of just 1 as before.

Comment (ntrel) [reply vlib/builtin/string.v:150]: @spytheman 2 string struct instances, but only one heap allocation. The temporary struct instance might get optimised by gcc.
